pennylane_lightning/lightning_qubit.py

```diff
@@ -70,17 +72,86 @@ def _chunk_iterable(it, num_chunks):
     return iter(lambda: tuple(islice(it, num_chunks)), ())
 
 
-def _remove_snapshot_from_operations(operations):
-    operations = operations.copy()
-    operations.discard("Snapshot")
-    return operations
-
-
-class LightningQubit(DefaultQubit):
+allowed_operations = {
+    "Identity",
+    "BasisState",
+    "QubitStateVector",
+    "QubitUnitary",
+    "ControlledQubitUnitary",
+    "MultiControlledX",
+    "DiagonalQubitUnitary",
+    "PauliX",
+    "PauliY",
+    "PauliZ",
+    "MultiRZ",
+    "Hadamard",
+    "S",
+    "Adjoint(S)",
+    "T",
+    "Adjoint(T)",
+    "SX",
+    "Adjoint(SX)",
+    "CNOT",
+    "SWAP",
+    "ISWAP",
+    "PSWAP",
+    "Adjoint(ISWAP)",
+    "SISWAP",
+    "Adjoint(SISWAP)",
+    "SQISW",
+    "CSWAP",
+    "Toffoli",
+    "CY",
+    "CZ",
+    "PhaseShift",
+    "ControlledPhaseShift",
+    "CPhase",
+    "RX",
+    "RY",
+    "RZ",
+    "Rot",
+    "CRX",
+    "CRY",
+    "CRZ",
+    "CRot",
+    "IsingXX",
+    "IsingYY",
+    "IsingZZ",
+    "IsingXY",
+    "SingleExcitation",
+    "SingleExcitationPlus",
+    "SingleExcitationMinus",
+    "DoubleExcitation",
+    "DoubleExcitationPlus",
+    "DoubleExcitationMinus",
+    "QubitCarry",
+    "QubitSum",
+    "OrbitalRotation",
+    "QFT",
+    "ECR",
```

Comment:
For the ops without direct C++ support, do they still decompose or have a fall-back ?

Reply:
The ops without direct C++ support fall back to the `applyMatrix` method. 
We apply the full matrix. Let's call this a first-order approach.